recipes/dd-handler.rb

```diff
@@ -73,4 +73,5 @@
   arguments [handler_config]
   supports :report => true, :exception => true
   action :nothing
-end.run_action(:enable) if node['datadog']['chef_handler_enable']
+  only_if { node['datadog']['chef_handler_enable'] }
```

Comment:
Thanks for pointing this out, I just verified that the guard is evaluated correctly at compile time, on both chef 11 and 12.

i.e with `node['datadog']['chef_handler_enable']` set to `false`, running `datadog:dd-handler` outputs the following:

```
Compiling Cookbooks...
[...]
Recipe: datadog::dd-handler
  * chef_gem[chef-handler-datadog] action install (up to date)
  * chef_handler[Chef::Handler::Datadog] action enable (skipped due to only_if)

Converging 3 resources
[...]
Recipe: datadog::dd-handler
  * chef_gem[chef-handler-datadog] action install (up to date)
  * chef_handler[Chef::Handler::Datadog] action nothing (skipped due to action :nothing)
```